librdmacm/examples/rstream.c

```diff
@@ -304,7 +304,7 @@ static void set_options(int fd)
 			rs_setsockopt(fd, SOL_RDMA, RDMA_INLINE, &inline_size,
 				      sizeof inline_size);
 		} else if (optimization == opt_bandwidth) {
-			val = 0;
+			val = 4;
```

Comment:
I'm soooo out of touch with this code...  One could argue the problem is actually in rs_create_ep() and not rstream.  The spot in rs_create_ep() where it does this:
```
    if (rs->cm_id->verbs->device->transport_type == IBV_TRANSPORT_IWARP)
        rs->opts |= RS_OPT_MSG_SEND;
```
adjusts the rsocket options, which causes the check below (also in rs_create_ep()) to fail:
```
    if ((rs->opts & RS_OPT_MSG_SEND) && (rs->sq_inline < RS_MSG_SIZE))
        return ERR(ENOTSUP);
```
Technically, there's nothing a use did wrong here.  I think a better fix is to handle this internally.  Something like:
```
    /* See comment for RS_OPT_MSG_SEND */
    if (rs->cm_id->verbs->device->transport_type == IBV_TRANSPORT_IWARP) {
        rs->opts |= RS_OPT_MSG_SEND;
        if (rs->sq_inline < RS_MSG_SIZE)
            rs->sq_inline = RS_MSG_SIZE;
    }
```

Reply:
Thanks for the feedback!